src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java

```diff
@@ -461,7 +462,12 @@ protected FeaturesInfo encodeSimpleFeatures(
                                     || value instanceof Float
                                             && ((Float) value) == Float.NEGATIVE_INFINITY) {
                                 jsonWriter.value("-Infinity");
-                            } else {
+                            }
+                            else if (value instanceof String && isJsonString((String) value)) {
```

Comment:
Oh no, that's going to slow down general JSON output generation too much (for all normal strings).
Instead, look into the attribute descriptor and look for the native type name, if available:

https://github.com/geotools/geotools/blob/e58d016c76e375653267c63eddbd17b5cbbf1a28/modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCFeatureSource.java#L289